core/modules/node/node.module

```diff
@@ -1901,6 +1909,33 @@ function node_block_view($delta = '', $settings = array()) {
         }
       }
       break;
+
+    case 'empty':
+      // Check if the promoted view exists and still has the block display.
+      $view = views_get_view('promoted');
+      if ($view && $view->access('block')) {
+        $nids = views_get_view_result('promoted', 'block');
+
+        if (empty($nids)) {
+          $default_message = '<p>' . t('No content has been promoted to the front page yet.') . '</p>';
```

Comment:
okay, but... We've changed it like 5 times :) THIS IS THE LAST TIME. (cause I have to update all the matching tests too).